src/test/util/setup_common.cpp

```diff
@@ -200,6 +199,13 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::ve
             strprintf("Invalid -socketevents ('%s') specified. Only these modes are supported: %s",
                       sem_str, GetSupportedSocketEventsStr()));
     }
+    {
+        auto stats_client = StatsdClient::make(*m_node.args);
+        if (!stats_client) {
+            throw std::runtime_error{strprintf("Cannot init Statsd client (%s)", util::ErrorString(stats_client).original)};
```

Comment:
The app should start whatever `stats_client` is initialized or not IMO.
It's not a critical failure.

nit: avoid code duplication between init.cpp & setup_common.cpp